datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -182,105 +517,33 @@ pub fn swap_reverting_projection(
-        let collect_left_threshold = config.hash_join_single_partition_threshold;
-        plan.transform_up(&|plan| {
-            let transformed = if let Some(hash_join) =
-                plan.as_any().downcast_ref::<HashJoinExec>()
-            {
-                match hash_join.partition_mode() {
-                    PartitionMode::Auto => {
-                        try_collect_left(hash_join, Some(collect_left_threshold))?
-                            .map_or_else(
-                                || partitioned_hash_join(hash_join).map(Some),
-                                |v| Ok(Some(v)),
-                            )?
-                    }
-                    PartitionMode::CollectLeft => try_collect_left(hash_join, None)?
-                        .map_or_else(
-                            || partitioned_hash_join(hash_join).map(Some),
-                            |v| Ok(Some(v)),
-                        )?,
-                    PartitionMode::Partitioned => {
-                        let left = hash_join.left();
-                        let right = hash_join.right();
-                        if should_swap_join_order(&**left, &**right)
-                            && supports_swap(*hash_join.join_type())
-                        {
-                            swap_hash_join(hash_join, PartitionMode::Partitioned)
-                                .map(Some)?
-                        } else {
-                            None
-                        }
-                    }
-                }
-            } else if let Some(cross_join) = plan.as_any().downcast_ref::<CrossJoinExec>()
-            {
-                let left = cross_join.left();
-                let right = cross_join.right();
-                if should_swap_join_order(&**left, &**right) {
-                    let new_join =
-                        CrossJoinExec::new(Arc::clone(right), Arc::clone(left));
-                    // TODO avoid adding ProjectionExec again and again, only adding Final Projection
-                    let proj: Arc<dyn ExecutionPlan> = Arc::new(ProjectionExec::try_new(
-                        swap_reverting_projection(&left.schema(), &right.schema()),
-                        Arc::new(new_join),
-                    )?);
-                    Some(proj)
-                } else {
-                    None
-                }
+fn swap_filter(filter: &JoinFilter) -> JoinFilter {
+    let column_indices = filter
+        .column_indices()
+        .iter()
+        .map(|idx| {
+            let side = if matches!(idx.side, JoinSide::Left) {
+                JoinSide::Right
             } else {
-                None
+                JoinSide::Left
             };
-
-            Ok(if let Some(transformed) = transformed {
-                Transformed::Yes(transformed)
-            } else {
-                Transformed::No(plan)
-            })
+            ColumnIndex {
+                index: idx.index,
+                side,
```

Comment:
```suggestion
                side: idx.side.negate(),
```
this would remove `let side` calculation above.